src/oops/SOLID/singleResponsibilityPrinciple/before/config/EmployeeConfig.java

```diff
@@ -0,0 +1,11 @@
+package oops.SOLID.singleResponsibilityPrinciple.before.config;
+
+import oops.SOLID.singleResponsibilityPrinciple.before.model.Employee;
+
+public class EmployeeConfig {
+
+    public String getRecordLocation(Employee employee) {
```

Comment:
This method can also be static.

Moreover, this method is highly coupled with FileRepository, doesn't it make sense to add it as instance method of Repository?

Reply:
This file has a dependency on Employee class, hence the coupling is with Employee, not with FileRepository. And once you have a lot more config than just the filePath, you'll eventually want to externalize it to application.properties or something, and a different config class helps do that easily following SRP.

And usually according to the best practices I've seen, methods are made static only when they are not operating on instances of other classes, and only depend on their arguments for the return value (like Math.sqrt).
That might be debatable here though. We could make it static.

Reply:
Makes sense! Thanks :)